<a href="https://colab.research.google.com/github/hirajya/CCDATSCL_EXERCISES_COM222/blob/main/Exercise5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exercise 5

In [58]:
import pandas as pd
import numpy as np

from scipy import stats
from scipy.stats import chi2_contingency


## Part 1: ANOVA

A teacher wants to investigate whether different teaching methods affect students’ exam scores. The dataset contains 50 students, each assigned to one of three teaching methods (A, B, or C).

Question: "Is there a difference in mean exam scores among students taught using three different teaching methods?"

- Independent variable (factor): Teaching Method (A, B, C)
- Dependent variable: Exam Score (numeric)

In [59]:
url = "https://github.com/robitussin/CCDATSCL_EXERCISES/blob/f56f00991116fcb717a8827d324aa6f599dea366/exam_scores.csv?raw=true"

df_exam_scores = pd.read_csv(url)
df_exam_scores


,StudentID,Method,Score
0,1,A,79
1,2,A,77
2,3,A,82
3,4,A,75
4,5,A,80
5,6,A,78
6,7,A,83
7,8,A,76
8,9,A,81
9,10,A,74


Calculate the number of students in each teaching method.



In [60]:
df_exam_scores = df_exam_scores.drop(columns=['StudentID'])

In [61]:
df_class = df_exam_scores.groupby('Method').count()
df_class

,Score
Method,
A,17
B,17
C,16


Compute the mean and standard deviation of exam scores for each group.

In [62]:
df_exam_scores.groupby('Method')['Score'].agg(['mean', 'std'])

,mean,std
Method,,
A,78.588235,2.575336
B,86.294118,2.365500
C,71.937500,1.768945


Use a plot to show the exam scores by teaching method.

In [63]:
# put your answer here

In [64]:
df_exam_scores.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Method  50 non-null     object
 1   Score   50 non-null     int64 
dtypes: int64(1), object(1)
memory usage: 932.0+ bytes


Perform the ANOVA Test
- Use scipy.stats.f_oneway to conduct a one-way ANOVA.
- Report the F-statistic and p-value.
- Using a significance level of α = 0.05, state whether you reject or fail to reject H₀.

In [65]:
group_a = df_exam_scores[df_exam_scores['Method'] == "A"]['Score']
group_b = df_exam_scores[df_exam_scores['Method'] == "B"]['Score']
group_c = df_exam_scores[df_exam_scores['Method'] == "C"]['Score']

f_val, p_val = stats.f_oneway(group_a, group_b, group_c)

Interpret the results. What does it say about the relationship between teaching method and exam scores?

In [66]:
print(f"F-Statistic: {f_val}, P-Value: {p_val}")


F-Statistic: 165.17845596593054, P-Value: 5.5032645988699085e-22


In [67]:
alpha = 0.05
if p_val < alpha:
  print("Significant difference found (Reject H0)")
else:
  print("No significant difference found (Fail to reject H0)")

Significant difference found (Reject H0)


With F(·,·) = 165.18 and p = 5.50 × 10⁻²² < 0.05, the null hypothesis is rejected. This suggests a statistically significant difference in mean exam scores among the teaching methods.


## Part 2: Chi-squared

A survey was conducted to see if there is a relationship between gender and littering behavior.

Question:
"Is gender associated with littering behavior?"

Each row represents one person.

In [68]:
url = "https://github.com/robitussin/CCDATSCL_EXERCISES/blob/f56f00991116fcb717a8827d324aa6f599dea366/litteringdata.csv?raw=true"

df_littering_data = pd.read_csv(url)
df_littering_data

,ID,Gender,Litter
0,1,Male,Yes
1,2,Male,Yes
2,3,Male,Yes
3,4,Male,Yes
4,5,Male,Yes
5,6,Male,Yes
6,7,Male,Yes
7,8,Male,Yes
8,9,Male,Yes
9,10,Male,Yes


In [69]:
df_littering_data = df_littering_data.drop(columns=['ID'])

Count the number of observations in each category for Gender and Litter separately.

In [70]:
df_littering_data[['Gender', 'Litter']].apply(pd.Series.value_counts)


,Gender,Litter
Female,20.0,NaN
Male,20.0,NaN
No,NaN,22.0
Yes,NaN,18.0


In [71]:
gender_counts = df_littering_data['Gender'].value_counts()

litter_counts = df_littering_data['Litter'].value_counts()

print("Gender counts:\n", gender_counts)
print("\nLitter counts:\n", litter_counts)


Gender counts:
 Gender
Male      20
Female    20
Name: count, dtype: int64

Litter counts:
 Litter
No     22
Yes    18
Name: count, dtype: int64


Using `pd.crosstab()`, create a contingency table showing counts of littering behavior for each gender.
- Write down the row totals, column totals, and grand total from the table.

In [72]:
contingency_table = pd.crosstab(df_littering_data['Gender'], df_littering_data['Litter'])
contingency_table


Litter,No,Yes
Gender,,
Female,14,6
Male,8,12


State the null hypothesis (H₀) and alternative hypothesis (H₁) for this chi-squared test.

In [73]:
chi2, p, dof, expected = chi2_contingency(contingency_table)

Use scipy.stats.chi2_contingency to calculate:
- Chi-squared statistic
- p-value
- Degrees of freedom

In [74]:
print(f"Chi-Square Statistic: {chi2:.4f}")
print(f"P-value: {p:.4f}")
print(f"Degrees of freedom: {dof:.4f}")

Chi-Square Statistic: 2.5253
P-value: 0.1120
Degrees of freedom: 1.0000


Using a significance level of `α = 0.05`, determine whether to reject or fail to reject H₀.


In [75]:
alpha = 0.05

Interpret your results. Is there evidence that gender and littering behavior are associated?

In [77]:
if p < alpha:
  print("Significant difference found (Reject H0)")
else:
  print("No significant difference found (Fail to reject H0)")

No significant difference found (Fail to reject H0)
